### PR TITLE
added a quick installer for windows

### PR DESCRIPTION
**Pull request by devilismyfriend**

### CODE REVIEW

File: codes/scripts/post_install.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    bnb_src = os.path.join(os.getcwd(), "resources/bitsandbytes_windows")
+    bnb_dest = os.path.join(sysconfig.get_paths()["purelib"], "bitsandbytes")
+    '''
+    cudnn_src = os.path.join(os.getcwd(), "resources/cudnn_windows")
+    #check if chudnn is in cwd
+    if not os.path.exists(cudnn_src):
+        print("Can't find CUDNN in resources, trying main folder...")
+        cudnn_src = os.path.join(os.getcwd(), "cudnn_windows")
+        if not os.path.exists(cudnn_src):
+            cudnn_url = "https://b1.thefileditch.ch/mwxKTEtelILoIbMbruuM.zip"
```
Comment (152334H) [inline codes/scripts/post_install.py:83]: what the fuck is this

File: Setup DLAS.bat [hunk 2]
```diff
@@ -0,0 +1,78 @@
+@echo off
+:: This file is part of sygil-webui (https://github.com/Sygil-Dev/sygil-webui/).
```
Comment (152334H) [inline Setup DLAS.bat:2]: Note that I would need to copy their license over as well for this.

File: .gitignore [hunk 3]
```diff
@@ -1,133 +0,0 @@
-#experiments/*
```
Comment (152334H) [inline .gitignore:1]: why did you delete this file?

File: codes/scripts/post_install.py (hunk 1 above)
Comment (152334H) [reply codes/scripts/post_install.py:83]: I am not going to merge code that downloads executable code from an arbitrary filehosting site. Please find a more reputable source for CUDNN installation.

File: codes/requirements.laxed.txt [hunk 4]
```diff
@@ -16,6 +16,9 @@ einops
 lambda-networks
 mup
 
+#UI
+customtkinter
+
```
Comment (152334H) [inline codes/requirements.laxed.txt:21]: i will probably separate this out to a 2nd file for windows specifically, or add some [windows] package installation tag

File: codes/scripts/post_install.py (hunk 1 above)
Comment (devilismyfriend) [reply codes/scripts/post_install.py:83]: I uploaded it there myself and it's the same link I use in my Stable tuner, if you have a website that you'd consider more reputable that can host the DLL files let me know, otherwise it can just be removed, it's meant to improve performance on 4000 cards.

File: Setup DLAS.bat (hunk 2 above)
Comment (devilismyfriend) [reply Setup DLAS.bat:2]: I can rewrite it, it's just a simple batch file

File: .gitignore (hunk 3 above)
Comment (devilismyfriend) [reply .gitignore:1]: I thought I just removed it from my commit, will amend

File: codes/requirements.laxed.txt (hunk 4 above)
Comment (devilismyfriend) [reply codes/requirements.laxed.txt:21]: ctk isn't windows only

**Review by 152334H — DISMISSED** [review]
just to indicate that this problem requires changes

File: codes/scripts/post_install.py (hunk 1 above)
Comment (152334H) [reply codes/scripts/post_install.py:83]: I will try to search for one

File: Setup DLAS.bat (hunk 2 above)
Comment (152334H) [reply Setup DLAS.bat:2]: Copying the license is fine, it's compatible, this is more of a note for myself to copy it as required

File: codes/requirements.laxed.txt (hunk 4 above)
Comment (152334H) [reply codes/requirements.laxed.txt:21]: Oh, my apologies.

Is the `Setup DLAS.bat` script supposed to be run on windows only, then? Wouldn't the venv be missing? Or are linux users okay with the normal installation approach?

Comment (devilismyfriend) [reply codes/requirements.laxed.txt:21]: Well this is just for windows currently, I dont have Linux installed currently to debug an SH file

File: codes/scripts/post_install.py (hunk 1 above)
Comment (152334H) [reply codes/scripts/post_install.py:83]: well this is disappointing

it seems like the last public version of cudnn that is not locked behind a login is 8.3.0.98

I will have to think about whether I should just merge it

Comment (152334H) [reply codes/scripts/post_install.py:83]: @devilismyfriend I found the url!

https://developer.download.nvidia.com/compute/redist/cudnn/v8.6.0/local_installers/11.8/cudnn-windows-x86_64-8.6.0.163_cuda11-archive.zip

This downloads cudnn from nvidia without any login. You might want to take it for your own repo as well :)

Comment (devilismyfriend) [reply codes/scripts/post_install.py:83]: Nice, I'll updtlate the link and make sure everything is going to the right places

**Comment by devilismyfriend** [comment]
updated the cudnn link to nvidia, restored the gitignore

**Comment by 152334H** [comment]
This is the blurb I plan to add to the readme:

---

### Windows Installation with Conda

This script by @devilismyfriend will install a conda environment named `DLAS` that uses a newer version of cuDNN for faster training:

```bat
git clone https://github.com/152334H/DL-Art-School
cd DL-Art-School
".\Setup DLAS.bat"
```

This requires that you have [miniconda](https://docs.conda.io/en/latest/miniconda.html)/[anaconda](https://www.anaconda.com/)  and git installed.

---

Would you like to change it, or add it yourself?

File: .gitignore [hunk 5]
```diff
@@ -131,3 +131,7 @@ venv.bak/
 
 # mypy
 .mypy_cache/
+experiments/CHANGEME_your_experiment_name/15022023_214111_EXAMPLE_gpt.yml
+*.0
+*.yml
+experiments/EXAMPLE_gpt.yml
```
Comment (152334H) [inline .gitignore:137]: I will probably change this again myself later to not include the *EXAMPLE_gpt.yml files here

File: codes/scripts/post_install.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+import subprocess
+from pathlib import Path
+import requests
+import zipfile
+if sys.version_info < (3, 8):
+    import importlib_metadata
+else:
+    import importlib.metadata as importlib_metadata
+
+req_file = os.path.join(os.getcwd(), "requirements.txt")
```
Comment (152334H) [inline codes/scripts/post_install.py:15]: I'm probably also going to clean up this file and remove redundant lines like these, unless you have objections

**Comment by devilismyfriend** [comment]
yeah sure, just some leftover code that could be useful later but not necessary

**Comment by 152334H** [comment]
notes to self:
- clean post_install file
- fix gitignore
- add info readme

if nothing else, i'll hit the merge button?

**Comment by devilismyfriend** [comment]
I'm working on a configurator to use that customtkinter but it will take a bit more time

**Comment by 152334H** [comment]
Ah, well

If you want me to keep this open until that's complete, I can do that

If alternatively you'd like to merge this now and make a 2nd PR later, just give a :+1: for this comment
